### DISCUSSION TITLE
How to provide adequate description for FCST_NPUT_VAR

### DISCUSSION THREAD

**Discussion by Ho-ChunHuang-NOAA** [dtcenter/METplus#1581] [Configuration] [answered]
I want to pair the FCST-VAR with OBS-VAR, and having difficulty to distinguish L1, Z1, 0-5000m,....etc.

I want to extract https://www.emc.ncep.noaa.gov/mmb/hchuang/methelp/rrfs.t12z.natlevf015.tm00.grib2
of record 1548, 1550, 1556, 1557, 1733.  Are the specific description  below correct/adequate?

FCST_VAR1_NAME = TMP
FCST_VAR1_LEVELS = Z2 or Z0-2

FCST_VAR2_NAME = DPT
FCST_VAR2_LEVELS = Z2 or Z0-2

FCST_VAR3_NAME = UGRD
FCST_VAR3_LEVELS = Z10 or Z0-10

FCST_VAR4_NAME = VGRD
FCST_VAR4_LEVELS = Z10 or Z0-10

FCST_VAR5_NAME = HPBL
FCST_VAR5_LEVELS = L0 or Z0

Just curious, for record 1691, how do we specify FCST_VAR5_LEVELS  for 30-0 mb above ground

**Comment by CPKalb** [comment, marked as answer]
Hi:

I am not able to view the included grib file.  Are you able to send this over to us?  Instructions for sending data can be found here: https://github.com/dtcenter/METplus/discussions/954

To specify levels, P is used for pressure levels whereas Z is used for heights in meters.  So, a 500mb pressure level would be specified as P500, while 2 meter temperature would have a level of Z2.  L is for a generic level.  To specify 30-0mb, you would want to use P30-0 for the level.

Without seeing your file, I'm guessing that you would want these:
FCST_VAR1_NAME = TMP
FCST_VAR1_LEVELS = Z2

FCST_VAR2_NAME = DPT
FCST_VAR2_LEVELS = Z2

FCST_VAR3_NAME = UGRD
FCST_VAR3_LEVELS = Z10

FCST_VAR4_NAME = VGRD
FCST_VAR4_LEVELS = Z10

I would need to see the file to determine what should be set for VAR5.

Christina

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, The file is in
/incoming/irap/met_help/huang_data/rrfs.t12z.natlevf033.tm00.grib2

Ho-Chun Huang, Ph.D.

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by CPKalb** [comment]
Hi:

This is what you want for records:
FCST_VAR1_NAME = TMP
FCST_VAR1_LEVELS = Z2 or Z0-2

FCST_VAR2_NAME = DPT
FCST_VAR2_LEVELS = Z2 or Z0-2

FCST_VAR3_NAME = UGRD
FCST_VAR3_LEVELS = Z10 or Z0-10

FCST_VAR4_NAME = VGRD
FCST_VAR4_LEVELS = Z10 or Z0-10

FCST_VAR5_NAME = HPBL
FCST_VAR5_LEVELS = L0

Thanks,
Christina

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, Christina


Thank you for your guidance specifying the VAR LEVEL.


Ho-Chun Huang, Ph.D.

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by Ho-ChunHuang-NOAA** [comment]
Please close the issue.

Ho-Chun Huang, Ph.D.

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by j-opatz** [comment]
Hi Ho-Chun,

I wanted to also comment saying I was able to access the file and check out the contents. In addition to what @CPKalb provided, you can access a specific record in the file using `R`:

> FCST_VAR1_NAME = TMP
FCST_VAR1_LEVELS = R1691

Using the `-` option with Z or P can provide unexpected errors, since you're attempting to grab a range of vertical or pressure levels. For example, trying to use plot_data_plane for TMP at Z2 (record 1548) but calling it with Z0-2 causes the following error:

`plot_data_plane rrfs.t12z.natlevf015.tm00.grib2 rr_test.ps 'name="TMP"; level="Z0-2";' -v 5`
`DEBUG 1: Opening data file: rrfs.t12z.natlevf015.tm00.grib2`
`DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_Gb2".`
`DEBUG 4: Switching the GRIB2 radius of the earth value of 6371.23 km to 6371.2 km for internal consistency.`
`DEBUG 4: `
`DEBUG 4: Lambert Conformal Grid Data:`
`DEBUG 4:    hemisphere: N`
`DEBUG 4:   scale_lat_1: 38.5`
`DEBUG 4:   scale_lat_2: 38.5`
`DEBUG 4:       lat_pin: 22.5742`
`DEBUG 4:       lon_pin: 121.702`
`DEBUG 4:         x_pin: 0`
`DEBUG 4:         y_pin: 0`
`DEBUG 4:    lon_orient: 97.5`
`DEBUG 4:          d_km: 13`
`DEBUG 4:          r_km: 6371.2`
`DEBUG 4:            nx: 393`
`DEBUG 4:            ny: 225`
`DEBUG 4:     so2_angle: 0`
`DEBUG 4: `
`DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 0, master table: 2, center: 7, local table: 1, parm_cat: 7, parm: 200`
`DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 0, master table: 2, center: 7, local table: 1, parm_cat: 7, parm: 200`
`DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - processed 1746 records from GRIB2 file: rrfs.t12z.natlevf015.tm00.grib2`
`DEBUG 4: VarInfoFactory::new_var_info() -> created new VarInfo object of type "FileType_Gb2".`
`WARNING: `
`WARNING: MetGrib2DataFile::data_plane() -> No matching record found for 'TMP/Z2-0'`
`WARNING: `
`ERROR  : `
`ERROR  : plot_data_plane -> trouble getting field "name="TMP"; level="Z0-2";" from file "rrfs.t12z.natlevf015.tm00.grib2"`
`ERROR  : `

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi,

Would I be able to find out what R1691 means from the user' guide?

Ho-Chun Huang, Ph.D.

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Comment by CPKalb** [comment]
Hi:

R specifies a grib record number.  It's preferable to use variable names and levels when possible.  That way, if the record numbers change, you will still get the correct variables.

Christina

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi,

Thank you for the R usage.  Please help me understand better.  The above
example confuses me a little bit.   For 2m TMP, I can use below as an
alternative if the record number of target variable always the same,

FCST_VAR1_NAME = TMP
FCST_VAR1_LEVELS = R1548

correct?

Ho-Chun Huang, Ph.D.

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958

**Reply by CPKalb** [reply]
Hi:

Yes, you can use the record number as you specified above.  However, it's typically more flexible to use the variable level when possible, which would be Z2 for temperature.

Tina

**Comment by Ho-ChunHuang-NOAA** [comment]
Thank you Tina, please close this issue.

Ho-Chun Huang, Ph.D.

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Rm. 2792

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3958
